src/qt/test/test_main.cpp

```diff
@@ -85,37 +85,45 @@ int main(int argc, char* argv[])
         setenv("QT_QPA_PLATFORM", "minimal", 0 /* overwrite */);
     #endif
 
-    BitcoinApplication app;
-    app.setOrganizationName(QAPP_ORG_NAME);
-    app.setApplicationName(QAPP_APP_NAME_DEFAULT "-test");
-    app.createNode(*init);
+
+    QCoreApplication::setOrganizationName(QAPP_ORG_NAME);
+    QCoreApplication::setApplicationName(QAPP_APP_NAME_DEFAULT "-test");
 
     int num_test_failures{0};
 
-    AppTests app_tests(app);
-    num_test_failures += QTest::qExec(&app_tests);
+    {
+        BitcoinApplication app;
```

Comment:
In commit "qt, test: Clean settings after tests" (0dcbad341b0a8420a899c6dce0db56dd0deaa036)

It seems reasonable to move the `BitcoinApplication app` into a nested scope, so BitcoinApplication is fully destroyed before `settings.clear()` is called. But I don't understand if this was a required change, or just cleanup, since I would expect the `settings.clear()` call to work in practice either way.

Reply:
It was a required change because `~BitcoinApplication` calls `~BitcoinGUI`, which in turn saves the window geometry to the settings.